core/src/test/java/com/linecorp/armeria/client/NoopDnsCache.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.net.UnknownHostException;
+import java.util.List;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+import io.netty.handler.codec.dns.DnsQuestion;
+import io.netty.handler.codec.dns.DnsRecord;
+
+/**
+ * A DNS cache that does nothing. This class is only used for testing.
+ */
+public enum NoopDnsCache implements DnsCache {
```

Comment:
Should we add `DnsCache.disabled()` instead?

Reply:
I intentionally did not add it to `DnsCache` because it conflicts with the minimum TLL which is 1 second.
Currently, users can't disable DNS cache by setting 0 to min TTL.
I think if we provide `DnsCache.disabled()`, we need to allow 0 min TTL as well.

Reply:
OK. Sounds reasonable.